### ISSUE TITLE
Information about possible setup

### ISSUE THREAD

**Issue by dwarcake** [OraOpenSource/apex-nitro#289]
### Issue Description
Currently we are working on a big project which is separated into 2 repositories.
We make use of the "Workspace images" to share our files between all different applications.

Repository 1:
Contains all general functionalities like authentication, authorization, navigation menu etc.
Here we have general JavaScript, image and SCSS files.
These are used in every application.

Repository 2:
Contains the business specific code and applications.
They use all JavaScript and SCSS files from repo 1 but have also their own files, which do not belong to repo 1.
These files are usually application / page specific files.

Is it possible to set up Nitro in such a way that the people that use repo 2 have the latest version of repo 1 + the files in repo 2?
I was thinking about a way to have 2 source folders in the same Nitro project, because each repository is checked out in a different folder.
I don't think this is possible at the moment, but is there another that we can achieve this situation?

A way I tried to solve this is by making repo 2 a fork of repo 1.
This way repo 2 can always pull in the latest changes from repo 1.
Unfortunately this is for alles files and not only the JavaScript,... files which makes the repository very large and hard to manage.

I hope the situation is clear enough for you :)

### APEX Nitro Version
Latest

### Operating System
Windows

### APEX Version
18.1

**Comment by dwarcake** [comment]
I think this setup is good if you have 2 applications: 1 general appl. and 1 with the business functionality.
However, currently we have around 10-15 general applications and 20-30 business applications (still growing every week).

There could be files in the business side we also want to share between those applications.
In which case we want both to use the "Workspace files" reference for both "types" of applications and both repositories.

Main reason being that we don't want to publish each application separately but just publish once and all files are accessible in every application.

Is that something which would be possible?

**Comment by vincentmorneau** [comment]
I think I don't understand your sharing logic then. Even if you have 10-15 general applications and 20-30 business applications, APEX still separates:
- Application files
- Workspace files

So my thinking is that you have a general repository for workspace files, which all application refer to with 
```
-- Hardcoded APEX Nitro URL for shared files
apex_application.g_company_images := '//localhost:4001/';
```

And each application point to their own respective files
```
apex_application.g_flow_images := '//localhost:4000/';
```

Can you explain how you would like APEX Nitro to support your scenario better?

**Comment by dwarcake** [comment]
I know it's a complex setup but I'll try the to explain as best as I can :-)

The main concern is that the workspace files can come from both repositories.

Example:
General repository provides a file which sets the background of the breadcrumb region to red.
This file is referenced in all applications, both general appl. and business appl.
_Note: the purpose of this file is that it can be used in different projects in the future which will all have the same base-functionalities/layout._

In our business applications for **our** project, the team wants to use a different background color, let's say green.
They want to add their own SCSS/JavaScript files in the workspace images so that they can add the green background to **all but only** the business applications.

In order to upload  the files per application you'd have to create 20-30 Nitro configurations it seems, no ?
+ you would need to switch between Nitro configs everytime you go to a different application?
The developer might switch several times a day between applications, which would make it hard to keep track of "which nitro config to run" I think.
Even if the application specific files are in the workspace files, as long as you don't reference them they won't be loaded.
So they can be both in the workspace files, but in different repositories.

In short what we want to achieve is something like:

- General repository point to workspace files
- Business repository points to workspace files _OR_ application files but not have all the overhead of different configurations

Maybe our "ideal" setup might not be possible with Nitro, just wanted your point of view on this situation :)
Either way we will keep using Nitro :)

**Comment by vincentmorneau** [comment]
I think I understand your setup. 

Can you think of a concrete way how APEX Nitro should change to support this? I would like to help but not sure how.

**Comment by dwarcake** [comment]
I think the easiest solution from a user perspective, don't know how much work it is to add this, would be to have multiple source folders.
Nitro could then take the files from the multiple source folders (in our case 2) and compile them all to 1 distributable folder (or separate?).

When publishing the same method can be used.
Take the source files from both folders, add them together and publish these to the workspace files.

This way it's possible to create 1 config with source folders:
- General repository
- Business repository

1 possible issue would be if there are files with the same name in both folders.
Possible solutions:
- Follow the order of the config where the last one will have priority (just replace the first file with the last one)
- Raise an error

**Comment by vincentmorneau** [comment]
Interesting idea, I like it. 

Will add it to the backlog but can't guarantee when I'll get to it. Thanks a lot!

**Comment by vincentmorneau** [comment]
with v5 this is now possible via ES module import / export.
